lib/tss/connector.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require 'httparty'
+require 'net/http'
+
+module TSS
+  # Utility class encapsulating synchronous communication with TSS.
+  class Connector
+    include HTTParty
+
+    base_uri TSS.configuration.base_uri
+
+    def self.auth_credentials
+      @auth_credentials ||= {
+        basic_auth: {
+          username: TSS.configuration.secret.freeze,
+          password: TSS.configuration.password.freeze
+        }
+      }.freeze
+    end
+
+    attr_reader :oid
+    def initialize(organization_id)
+      @oid = organization_id
+    end
+
+    # @param [String] - organization_id
+    # @raise [RuntimeError] TSS request failed
+    def organization
+      path = "/v1/organizations/#{oid}"
+      p path
+      response = self.class.authenticated_get(path)
+
+      handle_get_response(response, path, 'organization')
+    end
+
+    # @raise [RuntimeError] TSS request failed
+    def transactions
+      path = "#{base_path}/transactions"
+      response = self.class.authenticated_get(path)
+
+      handle_get_response(response, path, 'transactions')
+    end
+
+    # @raise [RuntimeError] TSS request failed
+    def transaction(transaction_id)
```

Comment:
I would reconsider the naming of methods like this one... maybe rename `#transaction` to `#get_transaction`, `#organization` to `#get_organization` and so on.